ibis/backends/sql/rewrites.py

```diff
@@ -110,6 +111,54 @@ def sort_to_select(_, **kwargs):
     return Select(_.parent, selections=_.values, sort_keys=_.keys)
 
 
+@replace(p.FillNa)
+def fillna_to_select(_, **kwargs):
+    """Rewrite FillNa to a Select node."""
+    if isinstance(_.replacements, Mapping):
+        mapping = _.replacements
+    else:
+        mapping = {
+            name: _.replacements
+            for name, type in _.parent.schema.items()
+            if type.nullable
+        }
+
+    if not mapping:
+        return _.parent
```

Comment:
I think the latter 2 would hit it. In other places we check for this in the method directly rather than in handling inside an op - I'll push a quick followup to clean this up. FWIW this rewrite rule was what we had in ibis 8.0, so this code has already existed in the wild for a while.

Reply:
Yep, just putting down thoughts based on the annotation.